### PR TITLE
Feat/xcomp 10406 phone input

### PR DESCRIPTION
**Pull request by 1izpena**
#### `🔍 Show`

**TASK**: 

### Description, Motivation and Context
Make MoleculePhoneInput use MoleculeInputField instead of AtomInput. This way, we get more features like different types of help texts. Also, we remove the unnecessary wrapper of AtomIcon.

### Types of changes

- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles
- [ ] 🛠️ Tool

### Screenshots - Animations
Screenshot 2023-09-26 at 12 06 13

### CODE REVIEW

**Review by xaviermariaconejo — APPROVED** [review]

File: components/molecule/phoneInput/src/index.scss [hunk 1]
```diff
@@ -37,15 +37,40 @@
...
+          left: 0;
         }
 
         &.splitted {
           border-radius: $bdrs-l;
           border: 1px solid $c-gray-light;
           width: 100%;
 
+          & .sui-AtomInput-input--noBorder {
+            border-width: 1px 0;
```
Comment (Alverd04) [inline components/molecule/phoneInput/src/index.scss]: I think there is a size for borders :3

File: components/molecule/phoneInput/src/index.scss [hunk 2]
```diff
@@ -37,15 +37,40 @@
...
+            border-style: solid;
+            border-color: $c-gray-light;
+          }
+
           &.sui-MoleculePhoneInput--error {
             border-color: $c-error;
+
+            & .sui-AtomInput-input--noBorder {
+              border-width: 1px 0;
```
Comment (Alverd04) [inline components/molecule/phoneInput/src/index.scss]: Same here

**Review by Alverd04 — APPROVED** [review]

**Review by Alverd04 — COMMENTED** [review]
Did we check the coverage for the component, and add some tests to check what happens if no help text is given, or if there's an error make sure the error text is renderes if given?

File: components/molecule/phoneInput/src/index.js [hunk 3]
```diff
@@ -102,28 +104,28 @@ export default function MoleculePhoneInput({
...
           )}
-          {visiblePrefixes && (
-            <AtomIcon
-              size={ATOM_ICON_SIZES.medium}
-              color={ATOM_ICON_COLORS.currentColor}
-            >
-              {showDropdown ? dropdownCloseIcon : dropdownIcon}
-            </AtomIcon>
-          )}
+          {visiblePrefixes && (showDropdown ? dropdownCloseIcon : dropdownIcon)}
```
Comment (andresin87) [inline components/molecule/phoneInput/src/index.js:107]:
```suggestion
          {visiblePrefixes && (
            <PrimitiveInjector
              size={ATOM_ICON_SIZES.medium}
              color={ATOM_ICON_COLORS.currentColor}
            >
              (showDropdown ? dropdownCloseIcon : dropdownIcon)
            </PrimitiveInjector>
            )}
```

There is a way to do not loose that configuration adding the functionality to let the user/consumer give its own Icon react component wrapper to the svg element

**Comment by 1izpena** [comment]
@Alverd04 Regarding your comment: _Did we check the coverage for the component, and add some tests to check what happens if no help text is given, or if there's an error, make sure the error text is rendered if given?_ I think it might not be necessary, as that responsibility belongs to `MoleculeInputField`, to which the handling of help texts is delegated, and it should be tested at the `MoleculeInputField` level as well.

File: components/molecule/phoneInput/src/index.scss [hunk 4]
```diff
@@ -37,15 +37,40 @@
 
         & > input {
           background: none;
+
+          &.sui-AtomInput-input--noBorder {
+            position: relative;
+          }
```
Comment (jordevo) [inline components/molecule/phoneInput/src/index.scss:43]: is it ok to be overriding styles this way? 🤔 shouldn't we allow for `atom/input` itself to behave as we want when required?

File: components/molecule/phoneInput/src/index.js (hunk 3 above)
Comment (1izpena) [reply components/molecule/phoneInput/src/index.js:107]: I think the problem is that the icons already come with a wrapper that includes that configuration, and it was being duplicated and overwritten. 
@Alverd04 could you confirm this? 🙇‍♀️

File: components/molecule/phoneInput/src/index.scss (hunk 4 above)
Comment (1izpena) [reply components/molecule/phoneInput/src/index.scss:43]: What change would you propose? 😄 
The issue is that `MoleculeInputField` wasn't designed to work within the context of `MoleculePhoneInput`, and its behaviour varies within this context. On its own, `MoleculeInputField` works well; it's a single input and occupies the same space, so they move in harmony. However, within `MoleculePhoneInput`, there are additional components that break this behaviour. So, I understand that `MoleculeInputField` shouldn't be aware of the issues that `MoleculePhoneInput` might cause. 😄

Comment (jordevo) [reply components/molecule/phoneInput/src/index.scss:43]: > So, I understand that `MoleculeInputField` shouldn't be aware of the issues that `MoleculePhoneInput` might cause. 😄

of course I don't mean any coupling or making `MoleculeInputField` aware of any wrapper / consumer. My point has more to do with potential problems down the road with this approach of overriding styles. If there's no other way around, fine, but we might find unexpected styles when theming for `MoleculeInputField`.

**Review by jordevo — APPROVED** [review]

**Review by ivanmlaborda — APPROVED** [review]

**Pull request merged**